### ISSUE TITLE
Last Suggestion not Showing Correctly

### ISSUE THREAD

**Issue by alexknop** [sspanak/tt9#315]
I have been seeing an issue on my Sonim XP5s (running Android 8.1) where the last item in the suggestions (the farthest right suggestion) is not completely showing (it is cut off on the end) when I first cycle through the suggestions. However, if I cycle through them a second time, by the second time it looks correct (see how "Hid" shows correctly the second time).
I will say I am not seeing this issue on my CAT S22 Flip which runs Android 11, so maybe something improved with Android over time.
I have poked at the code a bit, learned about RecyclerView, etc. The xml/layout looks good, and possibly that isn't where the issue lies because it eventually looks good on the second try. I'm thinking it could be something with the functions for selecting left/right. I would be curious where you think the issue may lie and I am open to debugging that specific code area myself:
![image](https://github.com/sspanak/tt9/assets/11878115/33596184-1114-46d1-b94e-34333bb2c7ac)
![image](https://github.com/sspanak/tt9/assets/11878115/085f13df-8ee2-4780-868b-18835c9d0483)

**Comment by alexknop** [comment]
also, I am noticing on first scroll, it doesn't scroll very smooth. It kind of shifts left and then back to the right. It's almost as if the left-most suggestion is left-aligned to the screen (see how "Gi" is aligned to the left of the screen) and it's not allowing it to be cut off a bit and fade off to the left. Hope that makes sense.

**Comment by sspanak** [comment]
Ah, I see. Sorry, for misinterpreting. 

However, scrolling works fine on both my phones. Selecting the previous/next word happens in `SuggestionsBar.scrollToSuggestion()`, but I see nothing wrong there. 

There are two possible problems:
1. It could be a RecyclerView bug in older Android versions (I'm testing on 10 and 11).
2. [People claim](https://stackoverflow.com/questions/36426129/recyclerview-scroll-to-position-not-working-every-time) sometimes the RecyclerView needs some time before it is able to scroll

I doubt it is the second case, but could please help with confirming? In `SuggestionsBar.scrollToSuggestion()` try replacing:
```java
mView.scrollToPosition(selectedIndex);
```
with this:
```java
new Handler().postDelayed(() -> mView.scrollToPosition(selectedIndex), 200);
```

Supposingly, scrolling will become inadequately slow, but it will work. Of course, this is not a solution, it is just a test.

**Comment by alexknop** [comment]
Ah, even without testing this code and scrolling slower, "Hid" popped up just fine. It does seem to be a speed thing.
Try pressing left when you first get suggestions (which goes immediately to the last suggestion). Is the last suggestion showing fine on your Android 10/11 phones?

**Comment by alexknop** [comment]
Yes, this workaround is working, which is bringing us closer to the root of the problem.
I've found that a more obvious way to test this bug is when you first get a list of suggestions, press the left key so that it goes directly to the last suggestion instead of scrolling left to right until the end. Typically when I press 1 and then the left key, I can't even see the "?" that is suggested, but I was able to with this patch

**Comment by sspanak** [comment]
There may be a better solution in that Stackoverflow thread. Instead of waiting for 200 ms, this is supposed to wait for the view to become ready, then scroll:

```
mView.post(() -> mView.scrollToPosition(selectedIndex));
```

Could you please replace the same line as above, but with the `.post()` alternative. It works very well on my phone, so if it fixes the problem on yours, we can keep it.

**Comment by alexknop** [comment]
That did not work very well, the delay works better. I was even able to reduce the delay to 100 ms and I don't notice any slowness, but get to see the last suggestion correctly.

**Comment by sspanak** [comment]
This is probably happening on other low end phones too, so I would like to have it fixed.

I have two ideas how to fix it, but before that, I need to know one more thing. But let's start with the original version of `SuggestionBar.java`, the one that's in `master`.

1. Type "4-4", then cycle the suggestions until you can see the last one.
2. Select it, but do not press OK.
3. Change the text case with "#" (or whatever key you have configured).

Are you still able to see the last suggestion? Has the list automatically scrolled to the beginning?

If yes, then `setSuggestions()` must also be fixed. Otherwise, we'll take care only of `scrollToSuggestion()`.

Also, could you please validate one more alternative way of fixing the problem? Since your phone needs some time to update the highlighted word before scrolling, let's reverse the order of actions and first scroll, then let it draw on the screen as long as it wants. This could again fail, but we could give it a try. So, `scrollToSuggestion()` would look like this:
```java
public void scrollToSuggestion(int increment) {
	if (suggestions.size() <= 1) {
		return;
	}

	int oldIndex = selectedIndex;

	selectedIndex = selectedIndex + increment;
	if (selectedIndex == suggestions.size()) {
		selectedIndex = 0;
	} else if (selectedIndex < 0) {
		selectedIndex = suggestions.size() - 1;
	}

	mView.scrollToPosition(selectedIndex);

	mSuggestionsAdapter.setSelection(selectedIndex);
	mSuggestionsAdapter.notifyItemChanged(oldIndex);
	mSuggestionsAdapter.notifyItemChanged(selectedIndex);
}
```

The other idea is to add a configuration option "alternative scrolling" or similar, that when enabled will allow scrolling with delay, which definitely works for you.

**Comment by sspanak** [comment]
That's so sad... OK, I guess I'll add a setting for the alternative scrolling method using a delay.
